File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int rev = 0;
        while(x != 0){
            int md = x % 10;

            // overflow check
            if(rev > INT_MAX/10 || rev < INT_MIN/10)
                return 0;

            rev = rev * 10 + md;
            x = x / 10;
        } 
        return rev;   
    }
};

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {

        int copy = x;
        int rev = 0;

        while(copy != 0){
            int ld = copy % 10;

            
            if(rev > Integer.MAX_VALUE/10 || rev < Integer.MIN_VALUE/10)
                return 0;

            rev = rev * 10 + ld;
            copy = copy / 10;      
        } 

        return rev;
    }
}